### PR TITLE
SE-98: fix export with `--dashboard` flag

### PR DESCRIPTION
**Pull request by pooknull**
https://perconadev.atlassian.net/browse/SE-98

This PR fixes the export with the `--dashboard` flag for a large number of dashboards.

Also, a `TestDashboard` test has been added to ensure that exporting with the `--dashboard` flag works well on all PMM dashboards.

### CODE REVIEW

**Review by svetasmirnova — CHANGES_REQUESTED** [review]
I am still getting error with the dashboard, attached to the bug report.

```
$ ./pmm-dump export --pmm-url='https://admin:svetasveta@172.17.0.3'  --allow-insecure-certs --dashboard='Percona - DB Cluster'
2024-06-05T23:49:28+03:00 INF Credential user was obtained from pmm-url
2024-06-05T23:49:28+03:00 INF Credential password was obtained from pmm-url
2024-06-05T23:49:28+03:00 FTL Error retrieving dashboard selectors: get selectors from dashboard: get selectors: parse templating vars: to prometheus var: not supported type by pmm-dump: constant

$ ./pmm-dump version
Version: v0.7.0-ga, Build: 7b4681f
```

**Comment by svetasmirnova** [comment]
Now I get:

```
$ ./pmm-dump export --pmm-url='https://admin:svetasveta@172.17.0.2'  --allow-insecure-certs --dashboard='Percona - DB Cluster'
2024-06-18T16:20:41+03:00 INF Credential user was obtained from pmm-url
2024-06-18T16:20:41+03:00 INF Credential password was obtained from pmm-url
2024-06-18T16:20:41+03:00 FTL Error retrieving dashboard selectors: get selectors from dashboard: get selectors: parse templating vars: parse templating var: parse templating query: interpolate query: failed to interpolate query
sveta@s76:~/src/percona/pmm-dump$ ./pmm-dump version
Version: v0.7.0-ga, Build: a486ff5

```

**Review by svetasmirnova — COMMENTED** [review]
pmm-dump now completes succesfully for the test dashboard. I still recommend adding a test for this case to avoid regressions in the future.

File: pkg/grafana/expr/vmquery.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+						s += "!~"
+					case f.IsNegative && !f.IsRegexp:
+						s += "!="
+					case !f.IsNegative && f.IsRegexp:
+						s += "=~"
+					case !f.IsNegative && !f.IsRegexp:
+						s += "="
+					}
+					s += fmt.Sprintf(`"%s"`, f.Value)
+				}
```
Comment (artemgavrilov) [inline pkg/grafana/expr/vmquery.go:58]: We can omit `else` here
```suggestion
				}
				
				s += f.Label
				switch {
				case f.IsNegative && f.IsRegexp:
					s += "!~"
				case f.IsNegative && !f.IsRegexp:
					s += "!="
				case !f.IsNegative && f.IsRegexp:
					s += "=~"
				case !f.IsNegative && !f.IsRegexp:
					s += "="
				}
				s += fmt.Sprintf(`"%s"`, f.Value)
				
```

**Review by artemgavrilov — APPROVED** [review]

**Review by svetasmirnova — APPROVED** [review]

**Pull request merged**